lib/ykpiv.c

```diff
@@ -246,95 +274,119 @@ ykpiv_rc ykpiv_list_readers(ykpiv_state *state, char *readers, size_t *len) {
   return YKPIV_OK;
 }
 
-ykpiv_rc ykpiv_transfer_data(ykpiv_state *state, const unsigned char *templ,
-    const unsigned char *in_data, long in_len,
-    unsigned char *out_data, unsigned long *out_len, int *sw) {
-  const unsigned char *in_ptr = in_data;
-  unsigned long max_out = *out_len;
-  ykpiv_rc res;
+static ykpiv_rc _begin_transaction(ykpiv_state *state) {
   long rc;
-  *out_len = 0;
+  ykpiv_rc res;
 
   rc = SCardBeginTransaction(state->card);
+  if((rc & 0xFFFFFFFF) == SCARD_W_RESET_CARD) {
```

Comment:
This masking is totally necessary, on the mac Sierra the size of type long seems a little bit different. The error code returned by SCardBeginTransaction isn't 0x80100068, it's 0xFFFFFFFF80100068, which caused a lot of confusing situations during my debug.